### PR TITLE
Food code fix

### LINKED ISSUE #8546: Can't bite fruits/veteables
(WEB REPORT BY: gringonius REMOTE: 158.69.55.2:7777)

> Revision (Should be above if you're viewing this from ingame!)
> bef905aad65ed3da2ed3b0d1b185921b2013dd84 
> 
> General description of the issue
> Trying to bite said plants off the floor gives you the "You have been attacked in the chest with the eggplant by [your character name]! message.
> 
> What you expected to happen
> For you to eat the plant, like how it happens with actual food items such as getmore food and chef recipes.
> 
> What actually happened
> What I described.
> 
> Steps to reproduce if possible
> Just bite a vegetable.

### LINKED ISSUE #11160: Force feeding balance
(WEB REPORT BY: zalo1302 REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 5a2fc473dce7735b867c6638ae5653029adc2e3f 
> 
> General description of the issue
> Force feeding is too easy and fast. Sometimes one is writing and already ate something that could have killed you. And forks are even worst.
> 
> What you expected to happen
> Some balance. More slow or something resistable. Give ideas.
> 
> What actually happened
> Got forcefeed amauri fruit and couldn't do anything about it. Like 2 seconds at most.
> Steps to reproduce if possible
> ....

### PR DESCRIPTION
**Pull request by unid15**
- Partially fixes #11160 by adding a delay to fork feeding (same as normal forcefeeding)
- Added can_consume() and consume() procs to food, made biting use them (fixes #8546)
- Slightly changed how trash items work - they now appear where the food item was (and not in your hands every time). Now, if you're eating spaghetti from a table with a fork, a plate will no longer appear in your hand when you're done

### CODE REVIEW

**Comment by Shadowmech88** [comment]
Is a fork delay necessary?

How many units of chems are transferred via a fork bite?

**Comment by unid15** [comment]
Exactly one bite's worth of reagents, so around 2-4 (depends on the food item)

**Comment by PJB3005** [comment]
Uh, there is already an `On_Consume` proc. You should merge the functionality.

**Comment by unid15** [comment]
It does something different though - it handles deletion of the food and creation of trash items

**Comment by PJB3005** [comment]
Actually it does more than that. Grown food overrides it to cause the eater pain when the fruit is thorny, for example.

You should definitely merge it, even if it's just because of the confusing naming.

**Comment by unid15** [comment]
Also it's used by forks

I think it would be better if they were separate

**Comment by PJB3005** [comment]
Then you should still rename it to have a less confusing name.

File: code/modules/reagents/reagent_containers/food/snacks.dm [hunk 1]
```diff
@@ -56,23 +57,38 @@
...
+
+			//If trash is a path (like /obj/item/banana_peel), create a new object
+			//If trash is an object, use the object
+
+			//If the food item was in somebody's hands when it was eaten, put the trash item into their hands
+			//Otherwise, put the trash item in the same place where the food item was
 			if(ispath(trash, /obj/item))
-				var/obj/item/TrashItem = new trash(user)
-				user.put_in_hands(TrashItem)
+				var/obj/item/TrashItem = new trash(old_loc)
```
Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Shouldnt this and the following line be putting the trash on the dropped loc and not the old loc

Comment (unid15) [inline code/modules/reagents/reagent_containers/food/snacks.dm:67]: Nope, it'll put the trash in the eater's hands as intended (unless the food was lying on the ground, in which case the trash will be there)

Comment (clusterfack) [inline code/modules/reagents/reagent_containers/food/snacks.dm]: Are you sure thats true in all cases? What if I'm feeding someone else?

**Comment by Intigracy** [comment]
@unid15 please **log** fork feeding as well as **the reagents inside of the food being eaten.**

Then merge this please.

File: code/game/objects/items/weapons/kitchen.dm [hunk 2]
```diff
@@ -78,7 +78,12 @@
...
 		else
-			user.visible_message("<span class='notice'>[user] feeds [M] a delicious forkful of [loaded_food_name]!</span>")
+			user.visible_message("<span class='notice'>[user] attempts to feed [M] a delicious forkful of [loaded_food_name].</span>")
+			if(do_mob(user, M))
+				if(!loaded_food)
+					return
+
+				user.visible_message("<span class='notice'>[user] feeds [M] a delicious forkful of [loaded_food_name]!</span>")
 		reagents.reaction(M, INGEST)
 		reagents.trans_to(M.reagents, reagents.total_volume)
```
Comment (9600bauds) [inline code/game/objects/items/weapons/kitchen.dm]: just like replace this with `reagents.trans_to(M.reagents, reagents.total_volume, log_transfer = TRUE, whodunnit = user)`

File: code/modules/reagents/reagent_containers/food/snacks.dm (hunk 1 above)
Comment (unid15) [inline code/modules/reagents/reagent_containers/food/snacks.dm:67]: Then the trash will appear in your hands, as it should

File: code/game/objects/items/weapons/kitchen.dm (hunk 2 above)
Comment (unid15) [inline code/game/objects/items/weapons/kitchen.dm]: done

**Comment by Intigracy** [comment]
ily pal

**Comment by 9600bauds** [comment]
Well it doesn't exactly work 100% like forcefeeding people, namely the messages
dreamseeker_2016-08-04_18-45-33
And forcefeeding goes to attack logs instead of chemistry logs I guess but it does work and should be clear

**Comment by Intigracy** [comment]
oh, it should probably be red at the least

**Pull request merged**